infra/embedding_step_functions/unified_embedding/handlers/line_polling.py

```
@@ -606,31 +638,40 @@ def _update_line_embedding_status_to_success(
-                target_line.embedding_status = EmbeddingStatus.SUCCESS.value
-                lines_by_receipt[image_id][receipt_id].append(target_line)
+                # Update embedding status for ALL lines in the visual row
+                for line in target_row:
+                    line.embedding_status = EmbeddingStatus.SUCCESS.value
+                    lines_by_receipt[image_id][receipt_id].append(line)
+
+                logger.debug(
+                    "Marked visual row lines as SUCCESS",
+                    primary_line_id=primary_line_id,
+                    row_line_count=len(target_row),
+                    row_line_ids=[l.line_id for l in target_row],
```

Ruff (0.14.13)

Ambiguous variable name: `l`

(E741)